pkg/cloud/services/ec2/instances.go

```diff
@@ -151,17 +151,25 @@ func (s *Service) CreateInstance(scope *scope.MachineScope) (*infrav1.Instance,
 	// TODO(vincepri): Move subnet picking logic to its own function/method.
 	if scope.AWSMachine.Spec.Subnet != nil && scope.AWSMachine.Spec.Subnet.ID != nil {
 		input.SubnetID = *scope.AWSMachine.Spec.Subnet.ID
-	} else if scope.AWSMachine.Spec.AvailabilityZone != nil {
-		sns := s.scope.Subnets().FilterPrivate().FilterByZone(*scope.AWSMachine.Spec.AvailabilityZone)
+	} else if scope.AWSMachine.Spec.AvailabilityZone != nil || scope.AWSMachine.Spec.FailureDomain != nil {
+		availabilityZone := scope.AWSMachine.Spec.AvailabilityZone
```

Comment:
Since `AvailabilityZone` is the deprecated field, should this be the fallback value instead of using it as the primary value here?

Reply:
Sure I can switch them up